base/base/defines.h

```diff
@@ -127,10 +127,10 @@
 /// because SIGABRT is easier to debug than SIGTRAP (the second one makes gdb crazy)
 #if !defined(chassert)
     #if defined(ABORT_ON_LOGICAL_ERROR)
-        #define chassert(x) static_cast<bool>(x) ? void(0) : abortOnFailedAssertion(#x)
+        #define chassert(x) static_cast<bool>(x) ? void(0) : ::DB::abortOnFailedAssertion(#x)
         #define UNREACHABLE() abort()
     #else
-        #define chassert(x) ((void)0)
+        #define chassert(x) (void)(x)
```

Comment:
It will evaluate `x` in release builds for all non-trivial `x` (when a compiler is not sure that the expression does not have side effects and can be safely optimized out)

Reply:
Yes, I assumed that it is OK, but since you are worrying let's avoid this for debug builds, fixed.